src/Http/SessionAuthenticator.php

```diff
@@ -15,19 +15,31 @@ class SessionAuthenticator
 {
     /**
      * @param Session $session
-     * @param int $userId
+     * @param AccessToken|int $token Token or user ID. Use of User ID is deprecated in beta 16, will be removed in beta 17
      */
-    public function logIn(Session $session, $userId)
+    public function logIn(Session $session, $token)
     {
+        // Backwards compatibility with $userId as parameter
+        // Remove in beta 17
+        if (! ($token instanceof AccessToken)) {
+            $token = AccessToken::generate($token);
+        }
+
         $session->regenerate(true);
-        $session->put('user_id', $userId);
+        $session->put('access_token', $token->token);
     }
 
     /**
      * @param Session $session
      */
     public function logOut(Session $session)
     {
+        $token = AccessToken::findValid($session->get('access_token'));
+
+        if ($token) {
```

Comment:
Should this clear all tokens for the user?

Reply:
My opinion is:

If you logout, it clear current session only.

In a future PR we add a "terminate all sessions" button in the session list. That option should also be offered when you change password. Similarly to Facebook which asks if you want to logout from all sessions or not on password change.